src/service-object.ts

```diff
@@ -430,7 +430,16 @@ class ServiceObject extends EventEmitter {
    * @param {string} reqOpts.uri - A URI relative to the baseUrl.
    * @param {function} callback - The callback function passed to `request`.
    */
-  request(reqOpts: DecorateRequestOptions): Promise<r.Response> {
+  request(reqOpts: DecorateRequestOptions): Promise<r.Response>;
+  request(reqOpts: DecorateRequestOptions, callback: r.RequestCallback): void;
+  request(reqOpts: DecorateRequestOptions, callback?: r.RequestCallback):
+      void|Promise<r.Response> {
+    if (callback) {
+      this.request_(reqOpts).then(
+          r => callback(null, r as r.Response, r.body),
```

Comment:
This line has a variable named `r` and a type named `r`. Can you remove the ambiguity even if the compiler / linter isn't complaining about it.